### DISCUSSION TITLE
Set S3 storage class when uploading with CLI

### DISCUSSION THREAD

**Discussion by toby1kenobi** [iterate-ch/cyberduck#16466] [Q&A]
On a Windows PC if I do this from a Powershell prompt the storage class is **not** set:

```
set "s3.storage.class=GLACIER_IR" & duck --username XXXX1234YYYYZZZZ --synchronize s3:/my-backups/ 'backups/' --existing upload
```

I guessed the set part by looking at a config file for the UI version of Cyberduck. Should the above work, or does anyone know what I should be doing?

**Comment by dkocher** [comment]
You will have to additionally run with the `--preserve` parameter to apply the storage class to the uploaded file.

**Reply by toby1kenobi** [reply]
Apologies, you suggested that on StackOverflow (thanks!) and I am actually doing that, the result is the same though. Here's what I actually try:

`set "s3.storage.class=GLACIER_IR" & duck --username XXXX1234YYYYZZZZ --preserve --synchronize s3:/my-backups/ 'backups/' --existing upload`

In Powershell I see this:

![Screenshot (114)](https://github.com/user-attachments/assets/8c167b37-4cbf-4288-a974-18f272cdc5c2)

then Cyberduck executes, but any files uploaded are Standard storage class:

![Screenshot (115)](https://github.com/user-attachments/assets/c03b80a4-a0b4-46c2-88d9-7412458c0f66)

Maybe I'm messing up something though?

**Reply by dkocher** [reply]
It works as advertised with the caveat that the storage class is preserved when overwriting an existing file.
